UI: 'pumblend' for cterm (256-color TUI)

justinmk opened the pull request:
From the ministry of silly walks.

- Seems to work OK. Would prefer to just have a "greyscale" effect for the negative space, but whatever.
- `hl_cterm2rgb_color` was adapted from Vim 8.1 (`cterm_color2rgb`).
- `hl_cterm2rgb_color` and `hl_rgb2cterm_color` seem useful for other cases, e.g. to add cterm support for cursor coloring? cc @teto 

ref #9571

justinmk committed:
UI: 'pumblend' for cterm (256-color TUI)

hl_rgb2cterm_color, hl_cterm2rgb_color were adapted from Vim 8.1
(color2index, cterm_color2rgb).
ref: https://github.com/vim/vim/commit/c5cd88554f1e0b2e9ff08d9a0748238dd8340ce1

teto commented:
I had written one `hl_rgb2cterm_color` but never merged it. Is that a patch from vim  ? yes it could relax the condition on termguicolors for cursorcolor IIRC

justinmk committed:
UI: change implementation of hl_rgb2cterm_color()

Replace the implementation cargo-culted from Vim's source with something
simpler which "seems to look better" with 'pumblend'.
